### ISSUE TITLE
[Feature] Pyright strict mode compatibility

### ISSUE THREAD

**Issue by jcheng5** [microsoft/playwright-python#1532]
Hi! I'm trying to use playwright with my codebase, which uses pyright in strict mode. I'm finding that assertions in the sync API make pyright unhappy, because `NoneType` turns into Unknown:

```python
# pyright: strict

from playwright.sync_api import Page, expect


def test_app(page: Page):
    page.goto("http://127.0.0.1:8000/")
    plot = page.locator("#plot")
    expect(plot).to_have_class("text-center")
```

```sh
$ pyright
No configuration file found.
No pyproject.toml file found.
stubPath /Users/jcheng/Development/jcheng5/nonetype-test/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
pyright 1.1.268
/Users/jcheng/Development/jcheng5/nonetype-test/test_playwright.py
  /Users/jcheng/Development/jcheng5/nonetype-test/test_playwright.py:9:5 - error: Type of "to_have_class" is partially unknown
    Type of "to_have_class" is "(expected: List[Pattern[str] | str] | Pattern[str] | str, *, timeout: float | None = None) -> Unknown" (reportUnknownMemberType)
1 error, 0 warnings, 0 informations
Completed in 0.634sec
```

If I replace all the `NoneType` annotations in `playwright/sync_api/_generated.py` with `None`, the error goes away.

(I just want to add, this is the only problem I've run into with playwright so far--in every other respect it's been a pleasure to use!)

**Comment by rwoll** [comment]
I'll raise with the team to better understand None v. NoneType choice here.

**Comment by jcheng5** [comment]
I've researched this as much as I can and came to the below conclusions, which I'm spelling out here on the off chance that it's helpful for your team. I also created PR #1536 in an effort to make this fix as low effort as possible.

`NoneType` is the type of the `None` object, but prior to Python 3.10 it was only internal. I am guessing `NoneType` ended up in the playwright APIs only because of how the Python annotation object represents itself as a string:

```python
>>> def foo() -> None: ...
>>> str(typing.get_type_hints(foo)["return"])
"<class 'NoneType'>"
```

And [this line in `process_type`](https://github.com/microsoft/playwright-python/blob/399f24fceb604afc3bbfb9a66a2ab02309c695fd/scripts/generate_api.py#L62) basically grabs whatever's in `<class '(.+?)'>`.

However, it's definitely not correct to actually use `NoneType` in your source code's type annotations; you're supposed to just use `None` instead. The playwright generated API files have to introduce a type alias for it (e.g. [here](https://github.com/microsoft/playwright-python/blob/399f24fceb604afc3bbfb9a66a2ab02309c695fd/playwright/sync_api/_generated.py#L87)).

The [mypy docs](https://mypy.readthedocs.io/en/stable/kinds_of_types.html) have this to say:

> The Python interpreter internally uses the name `NoneType` for the type of `None`, but `None` is always used in type annotations. The latter is shorter and reads better. (`NoneType` is available as [`types.NoneType`](https://docs.python.org/3/library/types.html#types.NoneType) on Python 3.10+, but is not exposed at all on earlier versions of Python.)
